### ISSUE TITLE
Search form accessibility

### ISSUE THREAD

**Issue by rshipp** [BanchanArt/banchan#547]
**Issue**: Buttons must have discernable text, inputs must have label
**Impact**: Critical
**Recommendation**: Add `role=”search”` to the `<form>` element here. Add `aria-label=”Search”` to the search button. Add `aria-label=”Search for offerings”` to the text input. Screenreaders do not announce the placeholder, so it's ok to duplicate that text in both places.
**More info**: https://dequeuniversity.com/rules/axe/4.6/button-name, https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Search_role 
**Screenshot**:
![image](https://github.com/BanchanArt/banchan/assets/1382374/11426eaf-9770-4f03-a4af-65b1687495e6)
